python/lsst/sphgeom/__init__.py

```diff
@@ -37,3 +38,130 @@
 from .version import *
 
 PixelizationABC.register(Pixelization)
+
+# Copy and paste from lsst.utils.wrappers
+_INTRINSIC_SPECIAL_ATTRIBUTES = frozenset(
+    (
+        "__qualname__",
+        "__module__",
+        "__metaclass__",
+        "__dict__",
+        "__weakref__",
+        "__class__",
+        "__subclasshook__",
+        "__name__",
+        "__doc__",
+    )
+)
+
+
+def _isAttributeSafeToTransfer(name: str, value: typing.Any) -> bool:
+    if name.startswith("__") and (
+        value is getattr(object, name, None) or name in _INTRINSIC_SPECIAL_ATTRIBUTES
+    ):
+        return False
+    return True
+
+
+def _continueClass(cls):
+    import sys
+
+    orig = getattr(sys.modules[cls.__module__], cls.__name__)
+    for name in dir(cls):
+        # Common descriptors like classmethod and staticmethod can only be
+        # accessed without invoking their magic if we use __dict__; if we use
+        # getattr on those we'll get e.g. a bound method instance on the dummy
+        # class rather than a classmethod instance we can put on the target
+        # class.
+        attr = cls.__dict__.get(name, None) or getattr(cls, name)
+        if _isAttributeSafeToTransfer(name, attr):
+            setattr(orig, name, attr)
+    return orig
+
+
+def _inf_to_lat(lat: float) -> float:
+    """Map +Inf to +90 and -Inf to -90 degrees."""
+    import math
+
+    if not math.isinf(lat):
+        return lat
+    if lat > 0.0:
+        return 90.0
+    return -90.0
+
+
+@_continueClass
+class Region:
+    """A minimal interface for 2-dimensional regions on the unit sphere."""
+
+    @classmethod
+    def from_ivoa_pos(cls, pos: str) -> Region:
+        """Create a Region from an IVOA POS string.
+
+        Parameters
+        ----------
+        pos : `str`
+            A string using the IVOA SIAv2 POS syntax.
+
+        Returns
+        -------
+        region : `Region`
+            A region equivalent to the POS string.
+
+        Notes
+        -----
+        See
+        https://ivoa.net/documents/SIA/20151223/REC-SIA-2.0-20151223.html#toc12
+        for a description of the POS parameter but in summary the options are:
+
+        * ``CIRCLE <longitude> <latitude> <radius>``
+        * ``RANGE <longitude1> <longitude2> <latitude1> <latitude2>``
+        * ``POLYGON <longitude1> <latitude1> ... (at least 3 pairs)``
+
+        Units are degrees in all coordinates.
+        """
+        shape, *coordinates = pos.split()
```

Comment:
Maybe do `strip` before `split` to protect from odd spaces?

Reply:
Doesn't look like I need to:
```python
>>> " a b  c        d       ".split()
['a', 'b', 'c', 'd']
```
so leading and trailing spaces are dropped anyhow.

Reply:
Ah, OK, my memory is not working well, it's probably only useful if you specify `sep` character.